mods/tnt/init.lua

```diff
@@ -411,6 +413,9 @@ minetest.register_node("tnt:gunpowder", {
 	on_punch = function(pos, node, puncher)
 		if puncher:get_wielded_item():get_name() == "default:torch" then
 			minetest.set_node(pos, {name = "tnt:gunpowder_burning"})
+			minetest.log("action", puncher:get_player_name() ..
+				" ignites " .. node.name .. " at " ..
```

Comment:
No need for 'node.name' here since it's always gunpowder, this can be simpler code with no concatenation.